src/auth_scheme/private_token.ts

```diff
@@ -317,14 +317,15 @@ export class Extension {
 
```

Comment:
for the line above



We should keep this check, and even harden it

```javascript
const MAXIMUM_EXTENSION_DATA_LENGTH = MAX_UINT-4
if (extensionData.length > MAXIMUM_EXTENSION_DATA_LENGTH) {
  throw new Error(`invalid extension data size. Max size is ${MAXIMUM_EXTENSION_DATA_LENGTH}.`);
}
```

It avoids discovering an extension is invalid at serialization time, while it could have been known when the constructor is called.